### ISSUE TITLE
Privacy Badger fails to import some settings

### ISSUE THREAD

**Issue by gitthepie** [EFForg/privacybadger#2488]
Privacy Badger's "Export user data" setting doesn't seem to work properly. I noticed 2 specific problems.

The first one is that "Prevent WebRTC from leaking local IP address" doesn't get imported properly. If you export the settings with this box checked, the checked box doesn't get imported properly into another browser. I tested with Firefox.

The second problem I noticed is that not all tracking domains get exported and imported properly. According to the "Manage Data" tab, lists of disabled sites are supposed to be combined. If this were true, the number of tracking domains detected by Privacy Badger should be the same on the source browser and the second browser importing the source browser's settings. However, I checked the number of domains on the second browser, and its number of "potential tracking domains" was lower than the number on the source browser, even after I restarted the second browser. Again, I did my test with Firefox.

**Comment by ghostwords** [comment]
Regarding tracking domains counts: please see https://github.com/EFForg/privacybadger/issues/2449.

Regarding WebRTC: it probably has to do with it being a special sort of setting, one that Privacy Badger doesn't have complete control over. For example, if another extension (or the user directly) already set the WebRTC setting, Privacy Badger will leave it alone. Does it sound like this is what's happening?

**Comment by ablanathtanalba** [comment]
I'm able to recreate the tracking domains count part of this issue, but only if I leave off the `www.` from the string of the disabled site. If I explicitly add `www.wired.com` to the list, export the data, then import it to a new badger instance, PB is properly disabled on that domain. I attempted the same thing with `wired.com`, and although the string itself would export/import properly, PB would still be enabled on the site if it didn't have the `www.` attached to it.

**Comment by gitthepie** [comment]
I read the other issue, and I see now that there is no actual impact to the number of trackers imported now. Still, not showing consistent numbers could lead to confusion for new Privacy Badger users.

The different numbers are also not helpful to users trying to remember which settings they have already imported settings to, or whether there are any new trackers on a device that usually imports settings rather than exporting settings. Maybe a user was using such a different device temporarily and wants those trackers to be exported to a device that usually exports rather than imports settings.

Keeping consistent numbers would help alleviate much confusion. At this point, I'm no longer concerned about Privacy Badger's effectiveness on this point; just its clarity in reporting numbers.

As for the failure to import the WebRTC setting, I do use another add-on (uBlock Origin) on both browsers. uBlock Origin has the ability to prevent WebRTC from leaking local IP addresses. However, I do not think uBlock Origin is the cause of the problem for two reasons.

First, I disabled the WebRTC feature on both uBlock Origin and Privacy Badger, then tried importing Privacy Badger settings again, but the box still remained unchecked in Privacy Badger. I also opened Privacy Badger's exported .json settings file in a text editor and found no evidence of a WebRTC setting, though I suppose I could have missed it.

Another possibility is that I set an about:config option in Firefox related to WebRTC that I forgot about. If Privacy Badger simply changes something in about:config, perhaps you can name the setting directly so I can check for it in the recipient browser.

**Comment by ghostwords** [comment]
>Still, not showing consistent numbers could lead to confusion for new Privacy Badger users.

Agreed! What if we only counted the base domains on that page? For example, the 46 `google.com` subdomains and `google.com` itself would be counted just once.

**Comment by ghostwords** [comment]
WebRTC: I see now. It is related to it being a special setting, but not the way I first considered. We don't save it in our settings object (we use the WebRTC setting browser API directly), and so as you noticed it doesn't get exported with the other settings.

We should either special-case for it on both export and import, or make it work like the rest of our settings while still respecting respecting overriding by other extensions/the browser itself.

**Comment by ghostwords** [comment]
Would you mind opening new tickets for these issues we uncovered?

**Comment by gitthepie** [comment]
I think counting only the base domains is a great idea! It's certainly better than the alternative; counting subdomains. Presumably, if we can't trust a domain, we generally can't trust its subdomains either.

So just to be clear, would these be reasonable ways to summarize the new tickets?:
1. Count only base domains in Tracking Domains
2. Properly export the setting "Prevent WebRTC from leaking local IP address" under "General Settings"

**Comment by ghostwords** [comment]
Sounds good. We can always rename later.

**Comment by ghostwords** [comment]
We could also keep this issue open and repurpose it for the Tracking Domains count discrepancy. Then, we only need one new issue (WebRTC setting ignored by user data export).

**Comment by gitthepie** [comment]
Done. The WebRTC issue has been added as [#2489](https://github.com/EFForg/privacybadger/issues/2489).

I suppose the current issue could be repurposed, but my initial title was vague because I did not know at the time that the current issue actually grouped two separate problems together. I added your idea of counting Tracking Domains consistently as [#2490](https://github.com/EFForg/privacybadger/issues/2490).

Should I close the current ticket or leave it open for further comment?
